TCC.Lib/Benchmark/BenchmarkHelper.cs

```diff
@@ -0,0 +1,251 @@
+
+            var iterations = GenerateBenchmarkIteration(benchmarkOption, data).ToList();
+
+            var operationSummaries = new List<OperationSummary>();
+
+            var threads = benchmarkOption.Threads == 0 ? Environment.ProcessorCount : benchmarkOption.Threads;
+
+            foreach (var iter in iterations)
+            {
+                PasswordMode pm = iter.Encryption ? PasswordMode.PublicKey : PasswordMode.None;
+                var compressedFolder = TestFileHelper.NewFolder();
+                var outputFolder = TestFileHelper.NewFolder();
+
+                // compress
+                var compressOption = new CompressOption
+                {
+                    Algo = iter.Algo,
+                    CompressionRatio = iter.CompressionRatio,
+                    BlockMode = BlockMode.Individual,
+                    SourceDirOrFile = iter.Content.Source,
+                    DestinationDir = compressedFolder,
+                    Threads = threads,
+                    PasswordOption = await BenchmarkOptionHelper.GenerateCompressPassswordOption(pm, keysFolder)
+                };
+
+                var swComp = Stopwatch.StartNew();
+                OperationSummary resultCompress = await _tarCompressCrypt.Compress(compressOption);
+                swComp.Stop();
+                if (_cancellationTokenSource.IsCancellationRequested)
+                {
+                    return null;
+                }
+
+                operationSummaries.Add(resultCompress);
+                foreach (var result in resultCompress.CommandResults)
+                {
+                    result.ThrowOnError();
+                }
+
+                var compressedSize = resultCompress.Blocks.Select(i => new FileInfo(i.DestinationArchive))
+                    .Select(f => f.Length)
+                    .Sum();
+
+                double compressionFactor = iter.Content.Size / (double)compressedSize;
+
+                // decompress
+                var decompressOption = new DecompressOption
+                {
+                    SourceDirOrFile = compressedFolder,
+                    DestinationDir = outputFolder,
+                    Threads = threads,
+                    PasswordOption = BenchmarkOptionHelper.GenerateDecompressPasswordOption(pm, keysFolder)
+                };
+                var swDecomp = Stopwatch.StartNew();
+                OperationSummary resultDecompress = await _tarCompressCrypt.Decompress(decompressOption);
+                swDecomp.Stop();
+                if (_cancellationTokenSource.IsCancellationRequested)
+                {
+                    return null;
+                }
+
+                operationSummaries.Add(resultDecompress);
+
+                foreach (var result in resultDecompress.CommandResults)
+                {
+                    result.ThrowOnError();
+                }
+
+                long comp = resultCompress.CommandResults.Sum(i => i.ElapsedMilliseconds);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused 'comp' local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1481)